### PR TITLE
Update to RxJava2

### PR DESCRIPTION
**Pull request by AshleyByeUK**
Updated as promised. I've not updated the docs. If you'd like me to do that, it may take a little more time. Just let me know.

### CODE REVIEW

**Comment by jponge** [comment]
Asking @tsegismont for review since he wrote that part.

**Comment by jponge** [comment]
@AshleyByeUK Yes eventually a text update is needed :-)

File: step-10/src/main/java/io/vertx/guides/wiki/database/WikiDatabaseServiceImpl.java [hunk 1]
```diff
@@ -109,40 +110,46 @@ public WikiDatabaseService fetchPageById(int id, Handler<AsyncResult<JsonObject>
...
+      .subscribe(SingleHelper.toObserver(resultHandler));
     return this;
   }
 
   @Override
   public WikiDatabaseService createPage(String title, String markdown, Handler<AsyncResult<Void>> resultHandler) {
     dbClient.rxUpdateWithParams(sqlQueries.get(SqlQuery.CREATE_PAGE), new JsonArray().add(title).add(markdown))
-      .map(res -> (Void) null)
-      .subscribe(RxHelper.toSubscriber(resultHandler));
+      .subscribe(
```
Comment (tsegismont) [inline step-10/src/main/java/io/vertx/guides/wiki/database/WikiDatabaseServiceImpl.java]: Same here, I think `toCompletable` and `CompletableHelper.toObserver` should do

File: step-10/src/main/java/io/vertx/guides/wiki/database/WikiDatabaseServiceImpl.java [hunk 2]
```diff
@@ -109,40 +110,46 @@ public WikiDatabaseService fetchPageById(int id, Handler<AsyncResult<JsonObject>
...
+      );
     return this;
   }
 
   @Override
   public WikiDatabaseService savePage(int id, String markdown, Handler<AsyncResult<Void>> resultHandler) {
     dbClient.rxUpdateWithParams(sqlQueries.get(SqlQuery.SAVE_PAGE), new JsonArray().add(markdown).add(id))
-      .map(res -> (Void) null)
-      .subscribe(RxHelper.toSubscriber(resultHandler));
+      .subscribe(
```
Comment (tsegismont) [inline step-10/src/main/java/io/vertx/guides/wiki/database/WikiDatabaseServiceImpl.java]: Same here, I think toCompletable and CompletableHelper.toObserver should do

File: step-10/src/main/java/io/vertx/guides/wiki/database/WikiDatabaseServiceImpl.java [hunk 3]
```diff
@@ -109,40 +110,46 @@ public WikiDatabaseService fetchPageById(int id, Handler<AsyncResult<JsonObject>
...
     return this;
   }
 
   @Override
   public WikiDatabaseService deletePage(int id, Handler<AsyncResult<Void>> resultHandler) {
     JsonArray data = new JsonArray().add(id);
     dbClient.rxUpdateWithParams(sqlQueries.get(SqlQuery.DELETE_PAGE), data)
-      .map(res -> (Void) null)
-      .subscribe(RxHelper.toSubscriber(resultHandler));
+      .subscribe(
```
Comment (tsegismont) [inline step-10/src/main/java/io/vertx/guides/wiki/database/WikiDatabaseServiceImpl.java]: Same here, I think toCompletable and CompletableHelper.toObserver should do

File: step-10/src/main/java/io/vertx/guides/wiki/http/HttpServerVerticle.java [hunk 4]
```diff
@@ -173,7 +176,7 @@ private void apiGetPage(RoutingContext context) {
 
   private void apiRoot(RoutingContext context) {
     dbService.rxFetchAllPagesData()
-      .flatMapObservable(Observable::from)
+      .flatMapObservable(Observable::fromIterable)
```
Comment (tsegismont) [inline step-10/src/main/java/io/vertx/guides/wiki/http/HttpServerVerticle.java]: Here we should favor a `Flowable`

File: step-8/src/main/java/io/vertx/guides/wiki/database/WikiDatabaseServiceImpl.java [hunk 5]
```diff
@@ -48,16 +49,16 @@
     this.sqlQueries = sqlQueries;
 
     getConnection()
-      .flatMap(conn -> conn.rxExecute(sqlQueries.get(SqlQuery.CREATE_PAGES_TABLE)))
-      .map(v -> this)
-      .subscribe(RxHelper.toSubscriber(readyHandler));
+      .map(conn -> conn.rxExecute(sqlQueries.get(SqlQuery.CREATE_PAGES_TABLE)).subscribe())
```
Comment (tsegismont) [inline step-8/src/main/java/io/vertx/guides/wiki/database/WikiDatabaseServiceImpl.java]: In this class you can apply all comments from step-10

File: step-8/src/main/java/io/vertx/guides/wiki/http/HttpServerVerticle.java [hunk 6]
```diff
@@ -192,16 +192,16 @@ public void start(Future<Void> startFuture) throws Exception {
       });
   }
 
-  private Single<Void> checkAuthorised(RoutingContext context, String authority) {
+  private Single<Boolean> checkAuthorised(RoutingContext context, String authority) {
```
Comment (tsegismont) [inline step-8/src/main/java/io/vertx/guides/wiki/http/HttpServerVerticle.java]: Use a `Completable` instead

File: step-8/src/main/java/io/vertx/guides/wiki/http/HttpServerVerticle.java [hunk 7]
```diff
@@ -266,7 +266,7 @@ private void apiGetPage(RoutingContext context) {
 
   private void apiRoot(RoutingContext context) {
     dbService.rxFetchAllPagesData()
-      .flatMapObservable(Observable::from)
+      .flatMapObservable(Observable::fromIterable)
```
Comment (tsegismont) [inline step-8/src/main/java/io/vertx/guides/wiki/http/HttpServerVerticle.java]: Favor a `Flowable` here

File: step-8/src/main/java/io/vertx/guides/wiki/http/HttpServerVerticle.java [hunk 8]
```diff
@@ -388,7 +388,7 @@ private void pageCreateHandler(RoutingContext context) {
 
   private void pageDeletionHandler(RoutingContext context) {
     checkAuthorised(context, "delete")
-      .flatMap(canDelete -> dbService.rxDeletePage(Integer.valueOf(context.request().getParam("id"))))
+      .map(canDelete -> dbService.rxDeletePage(Integer.valueOf(context.request().getParam("id"))))
```
Comment (tsegismont) [inline step-8/src/main/java/io/vertx/guides/wiki/http/HttpServerVerticle.java]: Use `flatMapCompletable`

File: step-9/src/main/java/io/vertx/guides/wiki/database/WikiDatabaseServiceImpl.java [hunk 9]
```diff
@@ -48,15 +49,15 @@
     this.sqlQueries = sqlQueries;
 
     getConnection()
-      .flatMap(conn -> conn.rxExecute(sqlQueries.get(SqlQuery.CREATE_PAGES_TABLE)))
-      .map(v -> this)
-      .subscribe(RxHelper.toSubscriber(readyHandler));
+      .map(conn -> conn.rxExecute(sqlQueries.get(SqlQuery.CREATE_PAGES_TABLE)).subscribe())
```
Comment (tsegismont) [inline step-9/src/main/java/io/vertx/guides/wiki/database/WikiDatabaseServiceImpl.java]: Comments from step-10 apply here as well

File: step-9/src/main/java/io/vertx/guides/wiki/http/HttpServerVerticle.java [hunk 10]
```diff
@@ -105,7 +105,7 @@ public void start(Future<Void> startFuture) throws Exception {
   private void apiDeletePage(RoutingContext context) {
     int id = Integer.valueOf(context.request().getParam("id"));
     dbService.rxDeletePage(id).subscribe(
-      v -> apiResponse(context, 200, null, null),
+      () -> apiResponse(context, 200, null, null),
```
Comment (tsegismont) [inline step-9/src/main/java/io/vertx/guides/wiki/http/HttpServerVerticle.java]: Comments from step-10 apply here

**Comment by jponge** [comment]
Any update on this front?

Do you guys think we could have it for the next update of the guide synchronized with 3.5.1?

Thanks!

**Comment by tsegismont** [comment]
@AshleyByeUK can you update the PR this week? Including Asciidoc changes?

**Comment by tsegismont** [comment]
@jponge as discussed offline, merged this after some more work in code/doc.

**Pull request merged**